### ISSUE TITLE
Pushnotification uses deprecated configuration and does not work with react-native 0.68 and gradle 7

### ISSUE THREAD

**Issue by Viktor-TechForm** [aws-amplify/amplify-js#9606]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Push Notifications

### Amplify Categories

Not applicable

### Environment information

<details>

```
  System:
    OS: macOS 12.2.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 1.46 GB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.14.0 - ~/Applications/homebrew/opt/node@16/bin/node
    Yarn: 1.22.17 - ~/Applications/homebrew/bin/yarn
    npm: 8.3.1 - ~/Applications/homebrew/opt/node@16/bin/npm
    Watchman: 2022.02.14.00 - /Users/vvf/Applications/homebrew/bin/watchman
  Browsers:
    Firefox: 97.0
    Safari: 15.3
  npmGlobalPackages:
    yarn: 1.22.17

```

</details>

### Describe the bug

React-native v0.68.0-rc1 upgrades gradle to version 7, which deprecates the `compile`dependency configuration: [upgrading from gradle 6 to 7](https://docs.gradle.org/current/userguide/upgrading_version_6.html#sec:configuration_removal).

This means that you can no longer build an android app with `@aws-amplify/pushnotification` installed.

### Expected behavior

I expected the android app to build.

### Reproduction steps

1. Init new react-native project with v0.68 or above
2. `npm i @aws-amplify/pushnotification`
3. `npm run android`

### Additional information and screenshots

Changing `compile` to `implementation` in `@aws-amplify/pushnotification/android/build.gradle` fixes the issue.

**Comment by sammartinez** [comment]
Thanks for reporting this @Viktor-TechForm, I will have someone from the team look at this today and get back to you once we are able to. Thanks!

**Comment by calebpollman** [comment]
Thanks for catching this @Viktor-TechForm!

Closing it in favor (and mark as a duplicate) of https://github.com/aws-amplify/amplify-js/issues/9537 as this issue happens happens on earlier versions of React Native as well.

**Comment by calebpollman** [comment]
@Simon-TechForm Thanks for the explanation and submitting the PR, it's appreciated!

**Comment by Simon-TechForm** [comment]
@calebpollman no problem :) This issue should probably be opened until a fix is merged, for others to see?

**Comment by calebpollman** [comment]
@Simon-TechForm  👍

**Comment by Viktor-TechForm** [comment]
Fixed by #9607
